.github/workflows/release-canary.yml

```diff
@@ -0,0 +1,33 @@
+name: release-canary
+
+on:
+  pull_request:
+    types: [labeled]
+
+jobs:
+  publish:
+    if: ${{ github.event.label.name == 'build' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      # Setup .npmrc file to publish to npm
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+          registry-url: 'https://registry.npmjs.org'
+      # Ensure that the README is published with the package
+      - run: rm -f packages/cli/README.md && cp README.md packages/cli
+      - run: npm ci --workspace packages/cli
+      - run: npm version $(npm run version --workspace packages/cli --silent)-canary-${{github.event.pull_request.number}}.$(git rev-parse --short HEAD) --workspace packages/cli
+      - run: npm publish --workspace packages/cli --tag canary-${{github.event.pull_request.number}}
```

Comment:
```suggestion
      - run: npm version 0.0.0-pr.${{github.event.pull_request.number}}.$(git rev-parse --short HEAD) --workspace packages/cli
      - run: npm publish --workspace packages/cli --tag experimental
```

While I get the idea behind making 1 tag per PR, making it easy for folks to always install the latest commit from a specific PR, it'll result in our NPM versions dashboard getting super cluttered and hard to find other tags. Also, for `pr` specific ones, I'd argue we should use the `0.0.0-pr.123.abc123` pattern, similar to the npm [preid](https://docs.npmjs.com/cli/v9/commands/npm-version#preid) flag's behaviour. 

The `canary` tag is usually considered specific type of release, like `alpha`/`beta`. I think it'd be smart if we left those open for ourselves and not use them up for PR preview releases. next-auth, React and others follow a similar formula.

Us:
![image](https://user-images.githubusercontent.com/7415984/233847963-a0f55939-3c6b-40b8-9c63-04ebf6af39b0.png)


Others: 
![image](https://user-images.githubusercontent.com/7415984/233847939-7d5581b0-c4c2-48b4-82d4-967d140b8501.png)
![image](https://user-images.githubusercontent.com/7415984/233848011-59d75a59-24ae-4809-ad75-24b5cf5caecf.png)